main.go

```diff
@@ -28,15 +33,58 @@ func main() {
 	}
 	logger.SetLevel(parsedLogLevel)
 
-	// prometheus conf
-	promMetricsHandler, _ := app.CreateMetricsHandler(logger, *namespace, *serviceName, *instanceName)
-	requestMetricName, _ := promMetricsHandler.CreateRequestsMetric()
+	if len(metricNames) == 0 {
+		panic(errors.New("at least one metric name should be given"))
+	}
+
+	// num_of_requests metric must exists since its metric handler contains the logic that makes the server a proxy,
+	// without it requests won't be forwarded to the forwardAddress
+	if !stringInSlice(string(metrics.NumOfRequestsMetricName), metricNames) {
+		metricNames = append(metricNames, string(metrics.NumOfRequestsMetricName))
+	}
+
+	var metricHandlers []metrics.MetricHandler
+	for _, metricName := range metricNames {
+		metricHandler, err := createMetricHandler(metricName, logger, *forwardAddress, *listenAddress, *namespace, *serviceName, *instanceName)
+		if err != nil {
+			panic(err)
+		}
+		metricHandlers = append(metricHandlers, metricHandler)
+	}
 
 	// proxy server start
-	proxyServer, err := app.CreateProxyServer(logger, *listenAddress, *forwardAddress,
-		promMetricsHandler, requestMetricName)
+	proxyServer, err := app.NewProxyServer(logger, *listenAddress, *forwardAddress, metricHandlers)
 	if err != nil {
 		logger.WithError(err).Fatal("Failed to create a proxy server")
 	}
-	proxyServer.Start()
+	if err = proxyServer.Start(); err != nil {
+		panic(err)
+	}
+}
+
+func createMetricHandler(metricName string,
+	logger *logrus.Logger,
+	forwardAddress string,
+	listenAddress string,
+	namespace string,
+	serviceName string,
+	instanceName string) (metrics.MetricHandler, error) {
+	switch metricName {
+	case string(metrics.NumOfRequestsMetricName):
+		return metrics.NewNumOfRequstsMetricsHandler(logger, forwardAddress, listenAddress, namespace, serviceName, instanceName)
+	case string(metrics.JupyterKernelBusynessMetricName):
+		return metrics.NewJupyterKernelBusynessMetricsHandler(logger, forwardAddress, listenAddress, namespace, serviceName, instanceName)
+	default:
+		var metricHandler metrics.MetricHandler
+		return metricHandler, errors.New("metric handler for this metric name does not exist")
```

Comment:
```suggestion
		return nil, errors.New("metric handler for this metric name does not exist")
```
Line 78 is not needed imo

Reply:
It's not a pointer can't be nil